python_modules/libraries/dagster-mariadb/dagster_mariadb/partitioned_io_manager_v2.py

```diff
@@ -0,0 +1,659 @@
+            return PartitionStrategy.daily_to_range(
+                partition_column=partition_column,
+                start_date=partitions_def.start,
+                end_date=partitions_def.end if hasattr(partitions_def, 'end') and partitions_def.end else "2099-12-31",
+            )
+        
+        elif isinstance(partitions_def, (WeeklyPartitionsDefinition, MonthlyPartitionsDefinition)):
+            # Treat weekly as daily for native partitioning (more granular)
+            return PartitionStrategy.monthly_to_range(
+                partition_column=partition_column,
+                start_date=partitions_def.start,
+                end_date=partitions_def.end if hasattr(partitions_def, 'end') and partitions_def.end else "2099-12-31",
+            ) if isinstance(partitions_def, MonthlyPartitionsDefinition) else PartitionStrategy.daily_to_range(
+                partition_column=partition_column,
+                start_date=partitions_def.start,
+                end_date=partitions_def.end if hasattr(partitions_def, 'end') and partitions_def.end else "2099-12-31",
+            )
+        
+        elif isinstance(partitions_def, HourlyPartitionsDefinition):
+            # For hourly, use daily partitioning (hourly is too granular)
+            return PartitionStrategy.daily_to_range(
+                partition_column=partition_column,
+                start_date=partitions_def.start,
+                end_date=partitions_def.end if hasattr(partitions_def, 'end') and partitions_def.end else "2099-12-31",
+            )
+        
+        elif isinstance(partitions_def, StaticPartitionsDefinition):
+            partition_keys = partitions_def.get_partition_keys()
+            partition_values = {key: [key] for key in partition_keys}
+            return PartitionStrategy.list_partitions(
+                partition_column=partition_column,
+                partition_values=partition_values
+            )
+        
+        else:
+            # Default to hash partitioning
+            return PartitionStrategy.hash_partitions(
+                partition_column=partition_column,
+                num_partitions=10
+            )
+
+    def _create_regular_table(
+        self,
+        connection,
+        context: OutputContext,
+        df: pd.DataFrame
+    ) -> None:
+        """Create a regular (non-partitioned) table."""
+        table_name = self._get_table_name(context)
+        
+        # Let pandas create the table
+        df.head(0).to_sql(
+            name=table_name,
+            con=connection,
+            schema=self.schema_name,
+            if_exists="fail",
+            index=False,
+        )
+        connection.commit()
+        context.log.info(f"Created regular table {self._get_full_table_name(context)}")
+
+    def _infer_sql_type(self, dtype) -> str:
+        """Infer SQL type from pandas dtype."""
+        dtype_str = str(dtype)
+        if 'int' in dtype_str:
```

Comment:
Int64 is case-sensitive and won't be caught by this (incorrect) heuristic. Correct way is to use introspection checks like is_integer_dtype etc.
https://pandas.pydata.org/docs/reference/arrays.html#data-type-introspection